src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala

```diff
@@ -48,16 +33,19 @@ object EasyFilesAndFolders {
 
   def getExistingParent(pathInDataset: String, datasetId: String): Try[Option[String]] = Try {
     val query: PreparedStatement = conn.prepareStatement("SELECT count(pid) FROM easy_folders WHERE path = ?")
-    val parentPath = pathInDataset
+    val subPaths = pathInDataset
       .split("/")
       .scanLeft("")((acc, next) => acc + next + "/")
       .reverse
-      .find(path => {
-          query.setString(1, path)
-          val resultSet = query.executeQuery()
-          if(!resultSet.next()) throw new RuntimeException("Count query returned no rows (?) A count query should ALWAYS return one row")
-          resultSet.getString("count") == "1"
-        })
+      .map(_.replaceAll("/$", ""))
+      .filter(!_.isEmpty)
+    val parentPath = subPaths.find(path => {
+        query.setString(1, path)
+        val resultSet = query.executeQuery()
+        // TODO don't forget to close the query?
```

Comment:
Not when you throw, but as the message says: it should not occur. Besides that it is a pity you loose the id here and have to fetch it later again. Neither have much priority at the moment.